integration/airflow/openlineage/airflow/extractors/python_extractor.py

```diff
@@ -0,0 +1,42 @@
+import os
+import inspect
+import logging
+from typing import Optional, List, Callable
+
+from openlineage.airflow.extractors.base import BaseExtractor, TaskMetadata
+from openlineage.client.facet import SourceCodeJobFacet
+
+
+log = logging.getLogger(__name__)
+
+
+class PythonExtractor(BaseExtractor):
+    @classmethod
+    def get_operator_classnames(cls) -> List[str]:
+        return ["PythonOperator"]
+
+    def extract(self) -> Optional[TaskMetadata]:
+        if os.environ.get("OPENLINEAGE_AIRFLOW_DISABLE_SOURCE_CODE"):
+            return None
+
+        source_code = self.get_source_code(self.operator.python_callable)
+        if source_code:
+            return TaskMetadata(
+                name=f"{self.operator.dag_id}.{self.operator.task_id}",
+                job_facets={
+                    "sourceCode": SourceCodeJobFacet(
+                        "python",
+                        # We're on worker and should have access to DAG files
+                        source_code
+                    )
+                }
+            )
+
+    def get_source_code(self, callable: Callable) -> Optional[str]:
+        try:
+            return inspect.getsource(callable)
```

Comment:
Inspect allows extracting code documentation as well (see `https://docs.python.org/3/library/inspect.html#inspect.getdoc`)
This could bring extra value to OpelLineage event.

Reply:
I'm kinda worried about getting garbage from default behavior:
```
If the documentation string for an object is not provided and the object is a class, a method, a property or a descriptor, retrieve the documentation string from the inheritance hierarchy.
```
```
or at the top of the Python source file (if the object is a module).
```